Website/src/rendering/src/components/Checkout/QuantityInput.tsx

```diff
@@ -0,0 +1,93 @@
+import { RequiredDeep } from 'ordercloud-javascript-sdk';
+import { DPriceSchedule } from 'src/models/ordercloud/DPriceSchedule';
+import { ChangeEvent, useState } from 'react';
+
+type QuantityInputProps = {
+  controlId: string;
+  priceSchedule: RequiredDeep<DPriceSchedule>;
+  disabled?: boolean;
+  quantity: number;
+  onChange: (quantity: number) => void;
+};
+
+const QuantityInput = (props: QuantityInputProps): JSX.Element => {
+  const [quantity, setQuantity] = useState(props.quantity);
+
+  const addDisabled =
+    props.disabled ||
+    (props.priceSchedule.MaxQuantity ? quantity >= props.priceSchedule.MaxQuantity : false);
+  const subtractDisabled =
+    props.disabled ||
+    (props.priceSchedule.MinQuantity ? quantity < props.priceSchedule.MinQuantity : quantity <= 0);
+
+  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    props.onChange(Number(e.target.value));
+  };
+
+  const handleSelectChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    props.onChange(Number(e.target.value));
+  };
+
+  const handleSubtract = () => {
+    const newQuantity = quantity - 1;
+    setQuantity(newQuantity);
+    props.onChange(newQuantity);
+  };
+
+  const handleAdd = () => {
+    const newQuantity = quantity + 1;
+    setQuantity(newQuantity);
+    props.onChange(newQuantity);
+  };
+
+  return (
+    <>
+      {props.priceSchedule.RestrictedQuantity ? (
+        <select
+          id={props.controlId}
+          disabled={props.disabled}
+          value={quantity}
+          onChange={handleSelectChange}
+        >
+          {props.priceSchedule.PriceBreaks.map((pb) => (
```

Comment:
We try to avoid short variable names that are not indicating their role like `s` and `li` in our projects. Could this one please be changed to `priceBreak`?